packages/py-api/py_api/controllers/hackathon_teams_controller.py

```diff
@@ -0,0 +1,124 @@
+import json
+import random
+import string
+from typing import Literal
+
+from bson.json_util import dumps
+from bson.objectid import ObjectId
+from fastapi.responses import JSONResponse
+from py_api.database.initialize import t_col
+from py_api.models import UpdateTeam
+from py_api.utilities.parsers import filter_none_values
+
+
+class TeamsController:
+    @classmethod
+    def create_team(cls, team_name: str, user_id: str) -> Literal[True] | Literal[False]:
+        """Creates a new team in the db with one team member.
+        This method should be used when a new participant is created.
+
+        Examples:
+
+        >>>result = participants_col.insert_one(participant_form_dump)
+        >>>TeamsController.create_team(participant_form_dump["team_name"], str(result.inserted_id))
+
+        Returns:
+             - True if the team was created
+             - False if such team exists
+        """
+
+        team_type = "normal"
+
+        if not team_name:
+            # Generates a random string of 8 characters if team_name is ""
+            # which means that the participant is random
+            team_name = ''.join(
+                random.choice(string.ascii_letters)
+                for _ in range(8)
+            )
+            team_type = "random"
+
+        team = t_col.find_one(filter={"team_name": team_name})
+
+        if not team:
+            t_col.insert_one({
+                "team_name": team_name,
+                "team_type": team_type,
+                "team_members": [user_id],
+            })
+            return True
+
+        return False
+
+    @classmethod
+    def append_team_members(cls, team_name: str, user_id: str) -> Literal[True] | Literal[False]:
+        team = t_col.find_one(filter={"team_name": team_name})
```

Comment:
This seems to be common functionality. Extract it in a function which accepts the team name as an argument and returns the team if found.